jetty-http/src/main/java/org/eclipse/jetty/http/HttpComplianceSection.java

```diff
@@ -0,0 +1,53 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2017 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.http;
+
+/**
+ */
+public enum HttpComplianceSection 
+{     
+    USE_CASE_INSENSITIVE_FIELD_VALUE_CACHE("","Use case insensitive field value cache"),
+    RFC7230_3_1_1_METHOD_CASE_SENSITIVE("https://tools.ietf.org/html/rfc7230#section-3.1.1","Method is case-sensitive"),
```

Comment:
This name is awkward.
What happens when RFC7230 is replaced?
I propose that the enum doesn't contain the RFC number and the section number, let the url have that.
Change it to the technology name of the spec.

Eg:

Current Name | New Name
-------------- | ------------
RFC7230_3_1_1_METHOD_CASE_SENSITIVE | HTTP_11_METHOD_CASE_SENSITIVE
RFC7230_3_2_FIELD_COLON | HTTP_11_FIELD_COLON
RFC7230_3_2_CASE_INSENSITIVE_FIELD_NAME | HTTP_11_CASE_INSENSITIVE_FIELD_NAME
RFC7230_3_2_4_NO_WS_AFTER_FIELD_NAME | HTTP_11_NO_WHITESPACE_AFTER_FIELD_NAME
RFC7230_3_2_4_NO_FOLDING | HTTP_11_NO_LINE_FOLDING
RFC7230_A2_NO_HTTP_9 | HTTP_09_NOT_SUPPORTED

Reply:
Good idea!  Actually do we even need HTTP_11_ prefix?

The other thing I've tossed around is should they be negations or enablers?  ie should it just be LINE_FOLDING and we don't add that for RFC7230 compliance.  That kind of feels better, but then the URL reference makes no sense as it points to a section that says it is not supported.

Reply:
I think it is worth including the technology name, just in case there is ever a change to specific functionality in the future where we have diverging but similar behaviors between, say, HTTP 1.1 and HTTP 2 that we need to differentiate from. I can't see this being common but feels like it might help future proof it some and possibly be easier to identify at a casual glance? Maybe I am being too pedantic.

Reply:
@WalkerWatch we should have a compliance page of its own that lists out all the options and what they represent

Reply:
@WalkerWatch problem with technology names is that it is hard to allocated. Some sections apply to just HTTP/1 (Eg white space after field name), some can apply to HTTP/1 and HTTP/2 (eg case insensitivity).    We also don't know how future specs will go, so we don't know if they will change.

Reply:
Our use of these is currently technology specific.
I can see someone wanting HTTP/1.1 to be more strict then HTTP/1.0 (for example), so having two different "levers" to pull would be a good thing.